datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -158,6 +159,13 @@ def assert_metric(self, name, value=None, tags=None, count=None, at_least=1, hos
         """
         Assert a metric was processed by this stub
         """
+
+        def _assert_metric(condition, msg):
```

Comment:
Let's refactor this into a method so it:

1. doesn't depend on scope
2. doesn't redefine a function for every call
3. can be used for service checks & events too

Reply:
Fixed, and added support for Service Checks.

Didn't implemented events, but will be easy to add later with the new refactored code.